monitoring/collector/etcd.go

```diff
@@ -0,0 +1,169 @@
+/*
+Copyright 2017 Gravitational, Inc.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package collector
+
+import (
+	"bytes"
+	"encoding/json"
+	"net/http"
+	"net/url"
+	"sync"
+
+	"github.com/gravitational/roundtrip"
+	"github.com/gravitational/satellite/monitoring"
+	"github.com/gravitational/trace"
+	"github.com/prometheus/client_golang/prometheus"
+)
+
+// ETCDCollector collects ETCD stats from the given server and exports them using
+// the prometheus metrics package.
+type ETCDCollector struct {
+	client *roundtrip.Client
+	config monitoring.ETCDConfig
+	mutex  sync.RWMutex
+
+	isRunning typedDesc
+	health    typedDesc
+	isLeader  typedDesc
+}
+
+// NewETCDCollector returns an initialized ETCDCollector.
+func NewETCDCollector(config *monitoring.ETCDConfig) (Collector, error) {
+	transport, err := config.NewHTTPTransport()
+	if err != nil {
+		return nil, trace.Wrap(err)
+	}
+
+	if len(config.Endpoints) == 0 {
+		return nil, trace.BadParameter("no ETCD endpoints configured")
+	}
+
+	client, err := newClient(config.Endpoints[0], roundtrip.HTTPClient(&http.Client{
+		Transport: transport,
+		Timeout:   collectMetricsTimeout,
+	}))
+
+	return &ETCDCollector{
+		client: client,
+		config: *config,
+		isRunning: typedDesc{prometheus.NewDesc(
+			prometheus.BuildFQName(namespace, "etcd", "up"),
+			"Whether scraping ETCD metrics was successful.",
```

Comment:
ETCD reads weird, can you replace with "etcd" everywhere? It's not an abbreviation.